src/ucs/sys/event_set.h

```diff
@@ -44,10 +45,14 @@ extern const unsigned ucs_sys_event_set_max_wait_events;
  * Allocate ucs_sys_event_set_t structure.
  *
  * @param [out] event_set_p  Event set pointer to initialize.
+ * @param [in]  user_fd      User defined event poll file descriptor. If -1 is
+ *                           specified, Event Set allocates new event poll
+ *                           file descriptor.
  *
  * @return UCS_OK on success or an error code on failure.
  */
-ucs_status_t ucs_event_set_create(ucs_sys_event_set_t **event_set_p);
+ucs_status_t
+ucs_event_set_create(ucs_sys_event_set_t **event_set_p, int user_fd);
```

Comment:
we need to define system-specific type for "native event set" which will be kqueue on BSD and int on Linux
it should be propagated up to UCP API and used instead of "int" for epfd
Also, add new function like ucs_event_set_create_from_epfd(

Reply:
@yosefe do you mean to update
`ucs_status_t ucp_worker_get_efd (ucp_worker_h worker, int *fd)`
in order to accept some wakeup handle (`int* -> ucp_wakeup_h`) ?

btw, `kqueue` returns file descriptor (`int`) like `epoll_create` does

Reply:
@dmitrygladkov in general yes, however since we can't change the API we can deprecate the above function, and extend ucp_worker_query()

Reply:
@yosefe thank you
I'll do it

btw, do we have some markers for deprecated APIs? to warn and/or document it?